explainer.md

```diff
@@ -0,0 +1,793 @@
+#### Instant() - constuctor
+
+#### Instant.prototype.epochSeconds : number
+
+The `epochSeconds` property of an `Instant` object is readonly and represents the *seconds
+since unix-epoch*.
+
+#### Instant.prototype.epochMilliseconds: number
+
+The `epochMilliseconds` property of an `Instant` object is readonly and represents the *milliseconds
+since unix-epoch*.
+
+#### Instant.prototype.epochMicroseconds: BigInt
+
+The `epochMicroseconds` property of an `Instant` object is readonly and represents the *microseconds
+since unix-epoch*.
+
+#### Instant.prototype.epochNanoseconds: BigInt
+
+The `epochNanoseconds` property of an `Instant` object is readonly and represents the *nanoseconds
+since unix-epoch*.
+
+#### Instant.prototype.witZone(zone: string) : ZonedDateTime
+
+This creates a `ZonedDateTime` by applying a *iana timezone* to the instant.
+
+This is equivalent to `new ZonedDateTime(instant, zone)`
+
+#### Instant.prototype.withOffset(offset: string) : OffsetDateTime
+
+This creates a `OffsetDateTime` by applying an *offset-string* to the instant.
+
+This is equivalent to `new OffsetDateTime(instant, offset)`
+
+#### Instant.fromString(iso: string) : Instant
+
+Parses a `string` that must be in the same ISO-8601 format as produced by `Instant.prototype.toString()` and creates a new `Instant` object from it.
+
+#### Instant.fromEpochSeconds(epochseconds: number) : Instant
+
+Equivalent to `Instant.fromMilliseconds(seconds * 1000)`.
+
+#### Instant.fromEpochMilliseconds(epochmillis: number) : Instant
+
+Equivalent to `Instant.fromEpochMicroseconds(BigInt(epochmillis) * 1000n)`.
+
+#### Instant.fromEpochMicroseconds(epochmicros: BigInt) : Instant
+
+Equivalent to `Instant.fromEpochNanoseconds(epochmicros * 1000n)`.
+
+#### Instant.fromEpochNanoseconds(epochnanos: BigInt) : Instant
+
+Equivalent to `new Instant(epochnanos)`.
+
+### OffsetDateTime
+
+An `OffsetDateTime` is an object that specifies a specific point in time with a specific offset from *UTC*.
+It bases this on an `Instant` and an offsett `string`.
+
+#### OffsetDateTime(instant: Instant, offset: string) - constructor
+
+The constructor may only be called as such. It takes two arguments. The first is an instance
+of `Instant` and the second is a `string` representing a valid time offset.
+
+#### OffsetDateTime.prototype.instant : BigInt
+
+The instance of `Instant` that was used to create this `OffsetDateTime`.
+
+#### OffsetDateTime.prototype.offset : string
+
+The time-zone offset represented as a `string`.
+
+This must always have the format: `sign``hours`:`minutes` where
+
+- `sign` is eitner `+` or `-`
+- `hours` is the hours offset 0-padded to 2 digits.
+- `minutes` is the minutes offset 0-padded to 2 digits.
+
+Examples: `+00:00`, `-04:00`, `+03:00`, ...
+
+#### OffsetDateTime.prototype.year: number
+
+The `.year` property represents the year of the `OffsetDateTime` according to the proleptic Gregorian calendar with a midnight to midnight 24 hour day.
```

Comment:
if it's necessary to specify "readonly" above, should it be present on all these as well?

Reply:
Moved to general section